components/widgets/settings/bool_setting.tsx

```diff
@@ -1,33 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
-import Setting from './setting.jsx';
+import Setting from './setting';
 
-export default class BoolSetting extends React.Component {
-    static propTypes = {
-        id: PropTypes.string.isRequired,
-        label: PropTypes.node.isRequired,
-        labelClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string.isRequired,
-        value: PropTypes.bool.isRequired,
-        inputClassName: PropTypes.string,
-        onChange: PropTypes.func,
-    };
+type BoolSettingProps = {
+    id: string;
+    label: React.ReactNode;
+    labelClassName: string;
+    helpText?: React.ReactNode;
+    placeholder: string;
+    value: boolean;
+    inputClassName: string;
+    onChange(name: string, value: any): void; // value is any since onChange is a common func for inputs and checkboxes
+}
 
-    static defaultProps = {
+export default class BoolSetting extends React.Component<BoolSettingProps> {
+    public static defaultProps: Partial<BoolSettingProps> = {
         labelClassName: '',
         inputClassName: '',
     };
 
-    handleChange = (e) => {
+    private handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
         this.props.onChange(this.props.id, e.target.checked);
     }
 
-    render() {
+    public render(): JSX.Element {
```

Comment:
nit: I don't know if we came up with the decision on which to use for consistency but some (including me) preferred to make this inferred only (less noise) but @devinbinnie can confirm.
```suggestion
    public render() {
```

Same case on other instance.

Reply:
@saturninoabril i added that according to https://github.com/mattermost/mattermost-webapp/pull/4061#discussion_r339570026 from @devinbinnie

Reply:
@saturninoabril Yeah I'm not sure if there was any decisions made about that, I just figured it adds more clarity.

Reply:
Thanks. I'm good with either approach, just a bit worried on consistency. Maybe should we start documenting style guide for Typescript, similar to what we have at https://docs.mattermost.com/developer/style-guide.html?